src/frontend/App/context/EkspanderbareVilkårpanelContext.tsx

```diff
@@ -0,0 +1,65 @@
+import constate from 'constate';
+import { useState } from 'react';
+
+export enum EkspandertTilstand {
+    EKSPANDERT = 'EKSPANDERT',
+    KOLLAPSET = 'KOLLAPSET',
+}
+
+export enum EInngangsvilkår {
+    MEDLEMSKAP = 'MEDLEMSKAP',
+    OPPHOLD = 'OPPHOLD',
+    MOR_ELLER_FAR = 'MOR_ELLER_FAR',
+    NYTT_BARN_SAMME_PARTNER = 'NYTT_BARN_SAMME_PARTNER',
+    SIVILSTAND = 'SIVILSTAND',
+    SAMLIV = 'SAMLIV',
+    ALENEOMSORG = 'ALENEOMSORG',
+}
+
+const [EkspanderbareVilkårpanelProvider, useEkspanderbareVilkårpanelContext] = constate(() => {
+    const allePanelEkspandert = Object.keys(EInngangsvilkår).reduce(
+        (acc, key) => ({
+            ...acc,
+            [key]: EkspandertTilstand.EKSPANDERT,
+        }),
+        {} as Record<EInngangsvilkår, EkspandertTilstand>
+    );
+
+    const [ekspanderteVilkår, settEkspanderteVilkår] =
+        useState<Record<EInngangsvilkår, EkspandertTilstand>>(allePanelEkspandert);
+
+    const toggleEkspandertTilstand = (key: EInngangsvilkår) => {
+        settEkspanderteVilkår({
+            ...ekspanderteVilkår,
+            [key]:
+                ekspanderteVilkår[key] === EkspandertTilstand.EKSPANDERT
+                    ? EkspandertTilstand.KOLLAPSET
+                    : EkspandertTilstand.EKSPANDERT,
+        });
```

Comment:
Når vi bruker forrige state for å utlede neste foretrekker jeg dette
```suggestion
        settEkspanderteVilkår((ekspanderteVilkår) => ({
            ...ekspanderteVilkår,
            [key]:
                ekspanderteVilkår[key] === EkspandertTilstand.EKSPANDERT
                    ? EkspandertTilstand.KOLLAPSET
                    : EkspandertTilstand.EKSPANDERT,
        }));
```
for å sikre at vi bruker nyeste state. Her har det ingen praktisk forskjell, og ref https://beta.reactjs.org/reference/react/useState#updating-state-based-on-the-previous-state er det ikke alltid nødvendig... Men jeg synes likevel det er greit å være konsekvent på det.